pkg/archive/archive_test.go

```diff
@@ -700,6 +701,56 @@ func tarUntar(t *testing.T, origin string, options *TarOptions) ([]Change, error
 	return ChangesDirs(origin, tmp)
 }
 
+func DecompressWithZstd(in []byte) []byte {
+	input := bytes.NewBuffer(in)
+	output := &bytes.Buffer{}
+	d, err := zstd.NewReader(input)
+	if err != nil {
+		return output.Bytes()
+	}
+	defer d.Close()
+	io.Copy(output, d)
+	return output.Bytes()
+}
+
+func CompressWithZstd(in []byte) []byte {
+	input := bytes.NewBuffer(in)
+	output := &bytes.Buffer{}
+	enc, _ := zstd.NewWriter(output)
+	io.Copy(enc, input)
+	enc.Close()
+	return output.Bytes()
+}
+
+func TestDetectCompression(t *testing.T) {
+	// test zstd compression without skippable frames.
+	compressedData := CompressWithZstd([]byte("hello world"))
+	compression := DetectCompression(compressedData)
+	if compression != Zstd {
+		t.Fatal("Unexpected compression")
+	}
+	// test zstd compression with skippable frames.
+	hex := []byte{
+		0x50, 0x2a, 0x4d, 0x18, // magic number of skippable frame: 0x184D2A50 to 0x184D2A5F
+		0x04, 0x00, 0x00, 0x00, // frame size
+		0x5d, 0x00, 0x00, 0x00, // user data
+		0x28, 0xb5, 0x2f, 0xfd, // magic number of Zstandard frame: 0xFD2FB528
+		0x04, 0x00, 0x31, 0x00, 0x00, // frame header
+		0x64, 0x6f, 0x63, 0x6b, 0x65, 0x72, // data block "docker"
+		0x16, 0x0e, 0x21, 0xc3, // content checksum
+	}
+	compression = DetectCompression(hex)
+	if compression != Zstd {
+		t.Fatal("Unexpected compression")
+	}
+
+	result := DecompressWithZstd(hex)
+	if string(result) != "docker" {
+		fmt.Println(result)
+		t.Fatal("Unexpected decompression result: " + string(result))
+	}
```

Comment:
Wondering what we're testing here; it looks like here we're testing that our test-data was  valid; if so, it should probably be validated _before_ the `DetectCompression(hex)` check above.

I'm a bit on the fence though; effectively we'd want a fixture here (this check would only fail if there was some change / bug in `github.com/klauspost/compress/zstd`), so I think it'd be fine to just (also see my other comment above) use a fixed value that we know is correct, and don't create / validate these on the fly.

Reply:
Actually, the purpose of this step is to ensure that the compressed data is in correct zstd format, so I decompress it and check the value. This section was removed.